packages/material-ui/src/Grid/Grid.js

```diff
@@ -74,11 +74,14 @@ function generateGutter(theme, breakpoint) {
       return;
     }
 
+    const gutter = theme.spacing(spacing / 2);
```

Comment:
This was my first approach to solve the problem, but I don't think that it's correct. f(x/2) != f(x)/2. We have no guarantee this property while hold. It does with a linear function, but it's not a generality. For instance, what if you return from an array index?

Reply:
In some cases maybe. But in the Grid component the spacing prop can takes only a single number, so that spacing function should return a single value too (accordingly TS specification).
In case when spacing function  takes an array of arguments:
```
const customSpacing = (...args: number[]) =>
  args.map(arg => `${arg * 0.5}rem`).join(' ');
```
it will works properly if Grid spacing provide some single number.
Or maybe I don't get it, and you can introduce the edge case when it's not a generality?

Reply:
Consider this custom spacing method:
```jsx
const customSpacing = x => [1,2,3,5,8,13,21][x];
```

Reply:
Oh, now I got it :) So it is better to use parseFloat() function maybe? And the result will look's like this
```
const themeSpacing = theme.spacing(spacing);

const spacingValue = parseFloat(themeSpacing);
const spacingDim = String(themeSpacing).replace(spacingValue, '') || 'px';
const gutter = spacingValue / 2;

const resultGutter = `${gutter}${spacingDim}`;
const resultWidth = `calc(100% + ${gutter}${spacingDim})`;
```
What do you think?

Reply:
Interesting. Which solution do you think is better between the parse float (here) and the regexp replace (a position solution posted on the issue)

Reply:
Well, there are some pros and cons on each other.
The regexp replace solution is more complex, and it should will handle string like 'rem123', but it's not correct measurement in CSS :). Need to change regexp - In current version it deals only with floating point numbers like '1.25 rem', but it's not working with integer like '3rem' (doesn't perform dividing). We have to perform a check that the themeSpacing is a String, for using .replace()  And finally it returns a string.
The parseFloat solution is simple as is :) It will return NaN if it can't parse values like 'rem123' (later we can use it to check in CSS string). Don't need to check typeof themeSpacing. And it returns a number.
So I think parseFloat solution is better. But not sure that I took into account all the cases.

Reply:
You are right about the integer support. The regexp would need to be updated to have the decimal support. I'm not aware of any more flaws. What about optimizing for the bundle size? Or maybe it doesn't matter? OK, pick what you think is best. We can always iterate on it in the future.

Reply:
Something is off with the visual regression tests.

Reply:
Regression test falls down because of changing the demo component. Now it ok.
The bundle size is slightly bigger, but it is 0.01%, I don't know is it matter. How this may affect the package in the future?